gtsam/geometry/tests/testPose2.cpp

```diff
@@ -835,6 +835,84 @@ TEST(Pose2 , ChartDerivatives) {
   CHECK_CHART_DERIVATIVES(T2,T1);
 }
 
+//******************************************************************************
+#include "testPoseAdjointMap.h"
+
+TEST(Pose2 , TransformCovariance3) {
+  // Use simple covariance matrices and transforms to create tests that can be
+  // validated with simple computations.
+  using namespace test_pose_adjoint_map;
+
+  // rotate
+  {
+    auto cov = GenerateFullCovariance<Pose2>({{0.1, 0.3, 0.7}});
+    auto cov_trans = RotateTranslate<Pose2>({{90.}}, {{}}, cov);
```

Comment:
I think we are now ready to move this to Lie.h :-) Might I suggest this should read
```
auto transformed = TransformCovariance(Pose2(0,0,90*degree))(cov);
```
where `TransformCovariance` is a more generic version of your RotateTranslate that works for any Lie group, and is defined as
```
template <class T>
class TransformCovariance {
private:
T::Jacobian adjointMap_;
public:
TransformCovariance(const T& X):adjointMap_(X.AdjointMap()) {}
operator ()(const T::Jacobian& covariance) {return adjointMap_ * covariance * adjointMap_.transpose();}
};
```
The "functor" form above makes it possible to transform many covariances without always having to re-compute the Adjoint map. In addition, we might want to consider templating on the Eigen matrix type so it works efficiently, say, with DiagonalMatrices.